pkg/testutil/testutil.go

```diff
@@ -681,31 +689,46 @@ func NewBaseWithNamespace(t *testing.T, ns string) *Base {
 	if ns == "" || ns == "default" || ns == Namespace {
 		t.Fatalf(`the other base namespace cannot be "%s"`, ns)
 	}
-	return newBase(t, ns, false)
+	return newBase(t, ns, false, false)
 }
 
 func NewBaseWithIPv6Compatible(t *testing.T) *Base {
-	return newBase(t, Namespace, true)
+	return newBase(t, Namespace, true, false)
+}
+
+func NewBaseForKube(t *testing.T) *Base {
+	base := newBase(t, "k8s.io", false, true)
+	// NOTE: kubectl namespaces are not the same as containerd namespaces.
+	// We still want kube test objects segregated in their own Kube API namespace.
+	KubectlHelper(base, "create", "namespace", Namespace).Run()
+	return base
 }
 
 func NewBase(t *testing.T) *Base {
-	return newBase(t, Namespace, false)
+	return newBase(t, Namespace, false, false)
 }
 
-func newBase(t *testing.T, ns string, ipv6Compatible bool) *Base {
+func newBase(t *testing.T, ns string, ipv6Compatible bool, kubeCompatible bool) *Base {
 	base := &Base{
 		T:                t,
 		Target:           GetTarget(),
 		DaemonIsKillable: GetDaemonIsKillable(),
 		EnableIPv6:       GetEnableIPv6(),
 		IPv6Compatible:   ipv6Compatible,
+		EnableKube:       GetEnableKube(),
+		KubeCompatible:   kubeCompatible,
 		Env:              os.Environ(),
 	}
 	if base.EnableIPv6 && !base.IPv6Compatible {
 		t.Skip("runner skips non-IPv6 compatible tests in the IPv6 environment")
 	} else if !base.EnableIPv6 && base.IPv6Compatible {
 		t.Skip("runner skips IPv6 compatible tests in the non-IPv6 environment")
 	}
+	if base.EnableKube && !base.KubeCompatible {
+		t.Skip("runner skips non-kube compatible tests in the kube environment")
```

Comment:
isn't it the opposite? 

`runner skips kube compatible tests in the kube environment` ?

Reply:
Arrrr, double negatives hell :D 

So, I guess what we have here is: `if base has KubeEnabled (kube environment), and if the test is not kube-compatible, we skip it`.

Seems correct?